2001-02-01  Richard Kenner  <kenner@vlsi1.ultra.nyu.edu>

	* config/i386/i386.c (ix86_compute_frame_size): Allow
	stack_alignment_needed to be non-default even if size is zero.

--- gcc/config/i386/i386.c.jj	2001/01/28 01:50:22	1.208
+++ gcc/config/i386/i386.c	2001/02/01 12:23:19	1.209
@@ -1886,12 +1886,11 @@ ix86_compute_frame_size (size, nregs_on_
 
   offset = frame_pointer_needed ? 8 : 4;
 
-  /* Do some sanity checking of stack_alignment_needed and preferred_alignment,
-     since i386 port is the only using those features that may break easilly.  */
+  /* Do some sanity checking of stack_alignment_needed and
+     preferred_alignment, since i386 port is the only using those features
+     that may break easilly.  */
 
   if (size && !stack_alignment_needed)
-    abort ();
-  if (!size && stack_alignment_needed != STACK_BOUNDARY / BITS_PER_UNIT)
     abort ();
   if (preferred_alignment < STACK_BOUNDARY / BITS_PER_UNIT)
     abort ();
